src/scribe_data/check/check_query_identifiers.py

```diff
@@ -0,0 +1,135 @@
+import re
+from pathlib import Path
+
+from scribe_data.cli.cli_utils import (
+    LANGUAGE_DATA_EXTRACTION_DIR,
+    language_metadata,
+    data_type_metadata,
+)
+
+
+def extract_qid_from_sparql(file_path: Path, pattern: str) -> str:
+    """
+    Extracts the QID from a SPARQL query file based on the provided pattern.
+
+    Parameters
+    ----------
+    file_path : Path
+        The path to the SPARQL query file from which to extract the QID.
+    pattern : str
+        The regex pattern used to match the QID (either for language or data type).
+
+    Returns
+    -------
+    str
+        The extracted QID if found, otherwise None.
+    """
+    try:
+        with open(file_path, "r", encoding="utf-8") as file:
+            content = file.read()
+            match = re.search(pattern, content)
+            if match:
+                return match.group(0).split("wd:")[1]
+    except Exception as e:
+        print(f"Error reading {file_path}: {e}")
+    return None
+
+
+def check_queries():
+    """
+    Validates SPARQL queries in the specified directory to check for correct language
+    and data type QIDs.
+
+    This function scans all SPARQL query files in the LANGUAGE_DATA_EXTRACTION_DIR
+    and prints out any files with incorrect QIDs for both languages and data types.
+    """
+    language_pattern = r"\?lexeme dct:language wd:Q\d+"
+    data_type_pattern = r"wikibase:lexicalCategory\s+wd:Q\d+"
```

Comment:
upgraded the regex expression

Reply:
and docstrings. Those were the major work in the commit.

are the docstrings in the right format? @andrewtavis

Reply:
Yes they're looking good now, @DeleMike :) I'd just indent parameter and return entries, but really good 😊